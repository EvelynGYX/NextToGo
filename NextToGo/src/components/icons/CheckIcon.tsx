import React from 'react';
import {StyleProp, ViewStyle} from 'react-native';
import FontAwesome5Icon from 'react-native-vector-icons/FontAwesome5';

export interface CheckIconProps {
  isChecked: boolean;
  onPress: () => void;
  style?: StyleProp<ViewStyle>;
}

const CheckIcon = (props: CheckIconProps) => {
  const isChecked = (isSelected: boolean): string => {
    if (isSelected) {
      return 'check-circle';
    } else {
      return 'circle';
    }
  };

  return (
    <FontAwesome5Icon
      style={props.style}
      name={isChecked(props.isChecked)}
      onPress={props.onPress}
      size={20}
    />
  );
};

export default React.memo(CheckIcon);
