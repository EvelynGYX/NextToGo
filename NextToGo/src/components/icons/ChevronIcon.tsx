import React from 'react';
import Entypo from 'react-native-vector-icons/Entypo';

export const ChevronRightIcon = (): JSX.Element => {
  return <Entypo name="chevron-right" size={18} />;
};
