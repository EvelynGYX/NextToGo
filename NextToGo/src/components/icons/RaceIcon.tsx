import React from 'react';
import {StyleProp, ViewStyle} from 'react-native';
import FontAwesome5Icon from 'react-native-vector-icons/FontAwesome5';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import Category from '../../constants/Category';

export interface RaceItemIconProps {
  category: Category;
  style?: StyleProp<ViewStyle>;
}

const RaceIcon = (props: RaceItemIconProps) => {
  return (
    <>
      {props.category === Category.Horse ? (
        <FontAwesome5Icon style={props.style} name="horse-head" size={20} />
      ) : props.category === Category.Greyhound ? (
        <FontAwesome5Icon style={props.style} name="dog" size={20} />
      ) : (
        <MaterialCommunityIcons
          style={props.style}
          name="horse-human"
          size={20}
        />
      )}
    </>
  );
};

export default RaceIcon;
