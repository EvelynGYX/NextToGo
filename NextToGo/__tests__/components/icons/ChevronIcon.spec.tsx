import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import {ChevronRightIcon} from '../../../src/components/icons/ChevronIcon';

describe('ChevronIcon', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ChevronRightIcon />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
