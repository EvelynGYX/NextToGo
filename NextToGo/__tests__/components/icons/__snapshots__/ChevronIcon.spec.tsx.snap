// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChevronIcon renders correctly 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 18,
      },
      undefined,
      Object {
        "fontFamily": "Entypo",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  ï…ƒ
</Text>
`;
